controllers/workspace/devworkspace_controller.go

```diff
@@ -142,6 +144,13 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	reqLogger = reqLogger.WithValues(constants.DevWorkspaceIDLoggerKey, workspace.Status.DevWorkspaceId)
 	reqLogger.Info("Reconciling Workspace", "resolvedConfig", configString)
 
+	// Inject ca certificates to the http clint if the certificates configmap is created and defined in the config.
+	if certs, ok := readCertificates(r.Client, config, r.Log); ok {
```

Comment:
2 observations:
1. We still want to use `config.GetGlobalConfig()` instead of `config` when calling `readCertificates()`, because the `config` struct might refer to an external DWOC that is used [only by a single workspace](https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#setting-an-alternate-configuration-for-a-workspace), yet, the global httpClient that is used by _all_ workspaces is being modified. Since multiple reconciles can happen in parallel, this could lead to a race condition where workspaces that do not configure SSL certs may end up using them when they are mid-reconcile. 

The `config *controller.OperatorConfiguration` parameter could thus be removed since we'd always be using  `config.GetGlobalConfig()` as the input, or it can remain. 

2. I think it might be worth to refactor this block of code that calls `readCertificates()` and `injectCertificates()` into an exported function `InjectCertificates()`, since we are calling the same things in `setupHttpClients()` and it'd be nice to hide some of these implementation details from the reconcile function. However, this is just my personal opinion, and not mandatory.

Reply:
I just want to double check, for an Eclipse Che installation, is the che-managed DWOC considered the global config?

Reply:
No, but very good point. The global DWOC is created in the namespace where DWO is installed whereas the Che-owned DWOC is created in the namespace where Che is installed.

My suggestion would limit us from letting Che configure the certs used from DWO. Instead, users will have to modify the global DWOC.

If we want to allow for external DWOC's, we should copy the http transport struct when injecting certs and use the copied one for the rest of the reconcile function to avoid modifying the global http transport object (and potentially causing a race condition).